addons/portal/portal_data.xml

```diff
@@ -7,5 +7,47 @@
             <field name="is_portal" eval="True"/>
         </record>
 
+        <record id="welcome_mail_portal_user" model="mail.template">
+            <field name="name">Welcome New Portal User</field>
+            <field name="model_id" ref="portal.model_portal_wizard_user"/>
```

Comment:
Change the model to the basic res_users model. This way you will be able to use object as user. In that case, you will have to put the portal_id in context.